crates/estree/src/util.rs

```diff
@@ -0,0 +1,97 @@
+use serde_json::Value;
+use serde::ser::*;
+
+pub enum Container {
+    NothingMeaningful,
+    FunctionDeclaration,
+    FunctionExpression,
+}
+
+/// A container used to represent some piece of data being
+/// serialized. Future versions may add options to the
+/// serialization format.
+pub struct Serialization<'a, T> where T: 'a {
```

Comment:
This is the file where I got stuck previous time. Can you please explain what are these containers doing / why are they necessary? Can't we just return own `json!()` generated structure and let consumer call `.serialize` on the target serializer they wish to serialize to?